test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -276,6 +290,56 @@ func Connect(mdb *mdbv1.MongoDB, opts *options.ClientOptions) error {
 	})
 }
 
+func ConnectAndVerifyExpectedCustomRoles(mdb *mdbv1.MongoDB, database, username, password string, expectedRoles []automationconfig.CustomRole) func(t *testing.T) {
+	return func(t *testing.T) {
+		if err := VerifyExpectedCustomRoles(t, mdb, database, expectedRoles, options.Client().SetAuth(options.Credential{
+			AuthMechanism: "SCRAM-SHA-256",
+			Username:      username,
+			Password:      password,
+		})); err != nil {
+			t.Fatalf("Error connecting to MongoDB deployment: %s", err)
+		}
+	}
+}
+
+func VerifyExpectedCustomRoles(t *testing.T, mdb *mdbv1.MongoDB, database string, expectedRoles []automationconfig.CustomRole, opts *options.ClientOptions) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+	mongoClient, err := mongo.Connect(ctx, opts.ApplyURI(mdb.MongoURI()))
+	if err != nil {
+		return err
+	}
+
+	return wait.Poll(time.Second*1, time.Second*30, func() (done bool, err error) {
+		var result bson.M
+		err = mongoClient.Database(database).
+			RunCommand(ctx, bson.D{
+				{"rolesInfo", 1},            // nolint
+				{"showPrivileges", true},    // nolint
+				{"showBuiltinRoles", false}, // nolint
+			}).Decode(&result)
+		if err != nil {
+			t.Fatal(err)
+			return false, nil
+		}
+
+		jsonRoles, err := json.Marshal(result["roles"])
```

Comment:
There should be a better way to do this than convert to json and back (since we have the info in bson). I'll try to use something like a `map[string][]CustomRole` in the `Decode()` call, since the result is a map with a roles field that contains all the result roles. Other ideas are welcome.

Reply:
Fixed!